contracts/Collector.sol

```diff
@@ -29,16 +32,23 @@ contract Collector is ICollector{
     external
     validShares(_partners)
     onlyOwner()
+    noRemainders()
     {
-        uint balance = token.balanceOf(address(this));
-        require(balance == 0, "can't update with balance > 0");
     
         delete partners;
         
         for (uint i = 0; i < _partners.length; i++)
             partners.push(_partners[i]);
     }
 
+    function updateRemainderAddress(address _remainderAddress) 
+    external
+    onlyOwner()
+    noRemainders()
```

Comment:
The idea is that the remainder address owns the right to any remainder balance that still remains in the contract before calling this function. This enforces that the remainderAddress can only be changed after a withdrawal. It is what made most sense to me but it's open to discuss.

Reply:
At the moment it's the owner of the contracts that allows the function to be called. Probably now that we introduced the `remainderAddress` it doesn't make sense this check either here or in the `updateShares` method. WDYT? After all, it's the owner that decide what to do, and with the multisig the partner should agree when changing the `remainderAddress`.

Reply:
After checking the code I think we may need to split the operations performed in the `noRemainders` modifier. Because we may probably want to send the remainder in some cases, and not sure if we want to check that there's no remainder